### PR TITLE
pull docker from their repos

### PR DESCRIPTION
**Pull request by davidsmejia**
## Issue Number

#954 

## Purpose/Implementation Notes

- installs docker-ce and docker-ce-cli from docker instead of ubuntu repository

## Types of changes

- Bugfix (non-breaking change which fixes an issue)

## Functional tests

n/a

## Checklist

- [X] Lint and unit tests pass locally with my changes

## Screenshots

n/a

### CODE REVIEW

File: infrastructure/api-configuration/api-server-instance-user-data.tpl.sh [hunk 1]
```diff
@@ -22,8 +22,10 @@ cp nginx.conf /etc/nginx/nginx.conf
 service nginx restart
 
 # install and run docker
-apt-get install docker-ce -y
-service docker restart
+apt-get install apt-transport-https ca-certificates curl software-properties-common -y
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu jammy stable" -y
```
Comment (arkid15r) [inline infrastructure/api-configuration/api-server-instance-user-data.tpl.sh:27]:
```suggestion
add-apt-repository "deb [arch=$(uname -m)] https://download.docker.com/linux/ubuntu jammy stable" -y
```

File: infrastructure/api-configuration/api-server-instance-user-data.tpl.sh [hunk 2]
```diff
@@ -22,8 +22,10 @@ cp nginx.conf /etc/nginx/nginx.conf
 service nginx restart
 
 # install and run docker
-apt-get install docker-ce -y
-service docker restart
```
Comment (arkid15r) [inline infrastructure/api-configuration/api-server-instance-user-data.tpl.sh:26]: I assume it does start automatically after the installation process is over?

**Review by arkid15r — APPROVED** [review]

**Comment by nozomione** [comment]
Just a thought, in the [Install Docker Engine on Ubuntu](https://docs.docker.com/engine/install/ubuntu) doc; 
- before installing the docker engine, it runs [`apt-get update`](https://www.cyberciti.biz/faq/what-does-sudo-apt-get-update-command-do-on-ubuntu-debian)(although not necessary).
- for adding GPG key, since [`apt-key` is deprecated](https://manpages.ubuntu.com/manpages/jammy/man8/apt-key.8.html)("*apt-key(8) will last be available in Debian 11 and Ubuntu 22.04.*" and writes a warning message to stdout), instead it uses the [gnupg](https://installati.one/ubuntu/22.04/gnupg/)'s [`gpg`](https://www.gnupg.org/documentation/manuals/gnupg/Operational-GPG-Commands.html) command. For this, perhaps it might be a good idea to use the gnupg?

(Additional resource) 
 [How To Handle apt-key and add-apt-repository Deprecation Using gpg to Add External Repositories on Ubuntu 22.04 - DigitalOcean](https://www.digitalocean.com/community/tutorials/how-to-handle-apt-key-and-add-apt-repository-deprecation-using-gpg-to-add-external-repositories-on-ubuntu-22-04)
 
 Thank you, David!

File: infrastructure/api-configuration/api-server-instance-user-data.tpl.sh (hunk 2 above)
Comment (davidsmejia) [reply infrastructure/api-configuration/api-server-instance-user-data.tpl.sh:26]: yeah it usually does this is just in case there is a reconcilable error

**Pull request merged**